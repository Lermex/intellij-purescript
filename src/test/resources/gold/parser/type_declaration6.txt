Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSTypeSynonymDeclarationImpl(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffects')
          PsiElement(whitespace)(' ')
        PSGenericIdentifierImpl(GenericIdentifier)
          PsiElement(identifier)('eff')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PsiElement(()('(')
            PsiElement(whitespace)(' ')
            PSRowImpl(Row)
              PSGenericIdentifierImpl(GenericIdentifier)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('random')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSTypeImpl(Type)
                PSTypeAtomImpl(TypeAtom)
                  PSTypeConstructorImpl(TypeConstructor)
                    PSQualifiedImpl(Qualified)
                      PSProperNameImpl(ProperName)
                        PsiElement(proper name)('RANDOM')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSGenericIdentifierImpl(GenericIdentifier)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSTypeImpl(Type)
                PSTypeAtomImpl(TypeAtom)
                  PSTypeConstructorImpl(TypeConstructor)
                    PSQualifiedImpl(Qualified)
                      PSProperNameImpl(ProperName)
                        PsiElement(proper name)('CONSOLE')
                        PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSTypeVarImpl(TypeVar)
                PSTypeConstructorImpl(TypeConstructor)
                  <empty list>
                PSGenericIdentifierImpl(GenericIdentifier)
                  PsiElement(identifier)('eff')
                  PsiElement(whitespace)(' ')
            PsiElement())(')')