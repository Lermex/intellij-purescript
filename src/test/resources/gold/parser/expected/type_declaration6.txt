Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('RandomConsoleEffects')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('eff')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('random')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PsiElement(proper name)('RANDOM')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(identifier)
                PsiElement(identifier)('console')
              PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PsiElement(proper name)('CONSOLE')
                      PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('eff')
                PsiElement(whitespace)(' ')
            PsiElement())(')')